toml/src/main/kotlin/org/rust/toml/inspections/CrateNotFoundInspection.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml.inspections
+
+import com.intellij.codeInspection.*
+import com.intellij.psi.PsiElementVisitor
+import org.rust.cargo.CargoConstants
+import org.rust.ide.experiments.RsExperiments
+import org.rust.openapiext.isFeatureEnabled
+import org.rust.toml.crates.local.CratesLocalIndexService
+import org.rust.toml.isDependencyKey
+import org.toml.lang.psi.*
+import org.toml.lang.psi.ext.TomlLiteralKind
+import org.toml.lang.psi.ext.kind
+
+class CrateNotFoundInspection : LocalInspectionTool() {
```

Comment:
I think this inspection should ignore dependencies with `path` or `git` specified? Seems like currently false-positive warning is shown instead. I mean something like `my_local_crate = { version = "1.0", path = "../my_local_crate" }` ([docs](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#multiple-locations))

Reply:
Also, looks like `TomlInlineTable` case is not covered by tests

Reply:
Well, I guess then it's back to having the `Dependency` class and parsing its properties? :sweat_smile: Otherwise I won't be able to check if it's a local/git dependency.

Reply:
Hm, probably 😄 
However please try to avoid holding `PsiElement`s. If it is necessary, `org.rust.openapiext.UtilsKt#createSmartPointer` might help to store a smart pointer instead

Reply:
Is it a problem in this case? I'm basically implementing a visitor which calls `visitDependency(object containing several PSI elements)`. It seems to me that it is the same situation as in normal PSI visitors.

Reply:
Oh, sorry, I looked at the code very briefly and didn't figure the problem out!
Looks like this warning is false-positive. I'll report this case to IntelliJ team a bit later. For now, you can either add `@Suppress("StatefulEp")` for `Dependency` class or rename it to something else (seems like it solves the warning), e.g. `DependencyCrate`

Reply:
BTW `DependencyCrate` and `DependencyTable` looks similar. Let's put them nearby and use the same naming for them, e.g. `val crateName: String, val crateElement: TomlKeySegment`.

`getDependencyTable` function could also be a factory method of `DependencyTable` named `from` or something like that, as far a I see.